lib/ecto/repo/queryable.ex

```diff
@@ -428,6 +465,25 @@ defmodule Ecto.Repo.Queryable do
     Query.where(queryable, [], ^Enum.to_list(clauses))
   end
 
+  defp query_for_reload([head| _] = structs) do
+    assert_structs!(structs)
+
+    schema = head.__struct__
+
+    case schema.__schema__(:primary_key) do
+      [pk] ->
+        keys = Enum.map(structs, &get_pk!(&1, pk))
+        Query.from(x in schema, where: field(x, ^pk) in ^keys)
+
+      pks ->
+        raise ArgumentError,
+              "Ecto.Repo.reload/2 requires the schema #{inspect(schema)} " <>
+                "to have exactly one primary key, got: #{inspect(pks)}"
+    end
+  end
+
+  defp query_for_reload(struct), do: query_for_reload([struct])
```

Comment:
I would remove this clause and call it as `query_for_reload([struct])` in the `reload`/`reload!` functions. :)